lib/features/steps/multipart_request_steps.rb

```diff
@@ -0,0 +1,126 @@
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the field {string} for multipart request is not null') do |part_key|
+  step %(the multipart field "#{part_key}" is not null)
+end
+
+# Tests that a multipart request field equals a string.
+#
+# @step_input part_key [String] The key to the multipart element
+# @step_input expected_value [String] The string to match against
+Then('the multipart field {string} equals {string}') do |part_key, expected_value|
+  parts = Server.current_request[:body]
+  assert_equal(parts[part_key], expected_value)
+end
+
+# (Deprecated) Retained for backwards compatibility
+# Tests that a multipart request field equals a string.
+#
+# @step_input part_key [String] The key to the multipart element
+# @step_input expected_value [String] The string to match against
+Then('the field {string} for multipart request equals {string}') do |part_key, expected_value|
+  step %(the multipart field "#{part_key}" equals "#{expected_value}")
+end
+
+# Tests that a multipart request field is null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the multipart field {string} is null') do |part_key|
+  parts = Server.current_request[:body]
+  assert_nil(parts[part_key], "The field '#{part_key}' should be null")
+end
+
+# (Deprecated) Retained for backwards compatibility
+# Tests that a multipart request field is null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the field {string} for multipart request is null') do |part_key|
+  step %(the multipart field "#{part_key}" is null)
+end
+
+# Tests that the multipart payload body does not match a JSON fixture.
+# JSON formatted multipart fields will be parsed into hashes.
+#
+# @step_input fixture_path [String] Path to a JSON fixture
+Then('the multipart body does not match the JSON fixture in {string}') do |fixture_path|
+  raw_payload_value = Server.current_request[:body]
+  payload_value = {}
+  raw_payload_value.each_with_object(payload_value) do |(k, v), out|
+    out[k] = JSON.parse(v.to_s)
+  rescue JSON::ParserError
+    out[k] = v.to_s
+  end
+  expected_value = JSON.parse(open(fixture_path, &:read))
+  result = value_compare(expected_value, payload_value)
+  assert_false(result.equal?, "Payload:\n#{payload_value}\nExpected:#{expected_value}")
+end
+
+# Tests that the multipart payload body matches a JSON fixture.
+# JSON formatted multipart fields will be parsed into hashes.
+#
+# @step_input fixture_path [String] Path to a JSON fixture
+Then('the multipart body matches the JSON fixture in {string}') do |fixture_path|
+  raw_payload_value = Server.current_request[:body]
+  payload_value = {}
+  raw_payload_value.each_with_object(payload_value) do |(k, v), out|
+    out[k] = JSON.parse(v.to_s)
+  rescue JSON::ParserError
+    out[k] = v.to_s
+  end
+  expected_value = JSON.parse(open(fixture_path, &:read))
+  result = value_compare(expected_value, payload_value)
```

Comment:
It might be worth extracting this to a function that this & the above step can share, if only the assertion is different